tests/mfem_ex9p_blockilu.cpp

```diff
@@ -0,0 +1,758 @@
+    6.8909427e-22, 1.3402368e-22, 7.041499e-23,  1.6868893e-23, 5.3817608e-24, 2.1628418e-24, 4.052847e-25,
+    2.077315e-25,  5.2805297e-26, 1.4991756e-26, 6.4111195e-27, 1.1854418e-27, 5.7778698e-28, 1.5775273e-28,
+    3.9627689e-29, 1.8055132e-29, 3.3888661e-30, 1.5252605e-30, 4.5133193e-31, 1.0063682e-31, 4.8541489e-32,
+    9.5035079e-33, 3.8425014e-33, 1.2400856e-33, 2.4894274e-34, 1.2507591e-34, 2.6116985e-35, 9.2835404e-36,
+    3.2802633e-36, 6.0873229e-37, 3.098741e-37,  7.0132001e-38, 2.1618231e-38, 8.3720174e-39, 1.490541e-39,
+    7.4015334e-40, 1.8352718e-40, 4.8812576e-41, 2.0657884e-41, 3.6793218e-42, 1.7084339e-42, 4.6726625e-43,
+    1.0773377e-43, 4.9368504e-44, 9.1410751e-45, 3.8190981e-45, 1.1567814e-45, 2.3503157e-46, 1.1444566e-46,
+    2.2689232e-47, 8.2874588e-48, 2.7850172e-48, 5.1392687e-49, 2.5770285e-49, 5.5785421e-50, 1.7509667e-50,
+    6.5244954e-51, 1.1413835e-51, 5.6431335e-52, 1.3495747e-52, 3.6180326e-53, 1.4884808e-53, 2.5915469e-54,
+    1.2030245e-54, 3.1993179e-55, 7.3638427e-56, 3.3095322e-56, 5.9941373e-57, 2.4996726e-57, 7.4165183e-58,
+    1.4925394e-58, 7.1770271e-59, 1.3975194e-59, 5.0699697e-60, 1.6799263e-60, 3.0561595e-61, 1.5190354e-61,
+    3.2472367e-62, 1.0062134e-62, 3.7179698e-63, 6.4089481e-64, 3.1397626e-64, 7.4558905e-65, 1.9623254e-65,
+    8.0421621e-66, 1.384426e-66,  6.3415152e-67, 1.6830301e-67, 3.7882438e-68, 1.7008593e-68, 3.0631187e-69,
+    1.2525424e-69, 3.7252989e-70, 7.3241056e-71, 3.518606e-71,  6.8539413e-72, 2.4224178e-72, 8.0767717e-73,
+    1.4398472e-73, 7.1226035e-74, 1.5313732e-74, 4.5983623e-75,
+};
+
+// Choice for the problem setup. The fluid velocity, initial condition and
+// inflow boundary condition are chosen based on this parameter.
+int problem;
+
+// Velocity coefficient
+void velocity_function(const Vector& x, Vector& v);
+
+// Initial condition
+double u0_function(const Vector& x);
+
+// Inflow boundary condition
+double inflow_function(const Vector& x);
+
+// Mesh bounding box
+Vector bb_min, bb_max;
+
+class DG_Solver : public Solver {
+private:
+  HypreParMatrix &      M, &K;
+  SparseMatrix          M_diag;
+  HypreParMatrix*       A;
+  serac::EquationSolver linear_solver;
+  double                dt;
+
+public:
+  DG_Solver(HypreParMatrix& M_, HypreParMatrix& K_, const FiniteElementSpace& fes)
+      : M(M_),
+        K(K_),
+        A(NULL),
+        linear_solver(M.GetComm(),
+                      serac::IterativeSolverParameters{.rel_tol     = 1e-9,
+                                                       .abs_tol     = 0.0,
+                                                       .print_level = 0,
+                                                       .max_iter    = 100,
+                                                       .lin_solver  = serac::LinearSolver::GMRES,
+                                                       .prec        = serac::BlockILUPrec{fes.GetFE(0)->GetDof()}}),
```

Comment:
The only functional change was replacing the GMRES/BlockILU members with a `serac::EquationSolver`